src/webgpu/shader/validation/uniformity/uniformity.spec.ts

```diff
@@ -0,0 +1,277 @@
+export const description = `Validation tests for uniformity analysis`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+export const kCollectiveOps = [
+  { op: 'textureSample', stage: 'fragment' },
+  { op: 'textureSampleBias', stage: 'fragment' },
+  { op: 'textureSampleCompare', stage: 'fragment' },
+  { op: 'dpdx', stage: 'fragment' },
+  { op: 'dpdxCoarse', stage: 'fragment' },
+  { op: 'dpdxFine', stage: 'fragment' },
+  { op: 'dpdy', stage: 'fragment' },
+  { op: 'dpdyCoarse', stage: 'fragment' },
+  { op: 'dpdyFine', stage: 'fragment' },
+  { op: 'fwidth', stage: 'fragment' },
+  { op: 'fwidthCoarse', stage: 'fragment' },
+  { op: 'fwidthFine', stage: 'fragment' },
+  { op: 'storageBarrier', stage: 'compute' },
+  { op: 'workgroupBarrier', stage: 'compute' },
+  { op: 'workgroupUniformLoad', stage: 'compute' },
+] as const;
+
+export const kConditions = [
+  { cond: 'uniform_storage_ro', expectation: true },
+  { cond: 'nonuniform_storage_ro', expectation: false },
+  { cond: 'nonuniform_storage_rw', expectation: false },
+  { cond: 'nonuniform_builtin', expectation: false },
+  { cond: 'uniform_literal', expectation: true },
+  { cond: 'uniform_const', expectation: true },
+  { cond: 'uniform_override', expectation: true },
+  { cond: 'uniform_let', expectation: true },
+  { cond: 'nonuniform_let', expectation: false },
+  { cond: 'uniform_or', expectation: true },
+  { cond: 'nonuniform_or1', expectation: false },
+  { cond: 'nonuniform_or2', expectation: false },
+  { cond: 'uniform_and', expectation: true },
+  { cond: 'nonuniform_and1', expectation: false },
+  { cond: 'nonuniform_and2', expectation: false },
+  { cond: 'uniform_func_var', expectation: true },
+  { cond: 'nonuniform_func_var', expectation: false },
+] as const;
+
+export function generateCondition({ condition }: { condition: string }) {
```

Comment:
As below, I am confused by the use of structured input here, and also I think a return type should be specified.

```
function generateCondition(condition: string): string
```